File: rails_demo/app/channels/message_channel.rb
class MessageChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from 'MessageChannel'
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    ActionCable.server.broadcast 'MessageChannel', {message: data['message']}
  end
end

File: rails_demo/app/javascript/channels/message_channel.js
import consumer from "channels/consumer";

consumer.subscriptions.create("MessageChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    this.perform("speak", {
      message: "I just connected, how's it goin' ya'll?",
    });
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log(data);
  },
});
